envoy/http/filter.h

```diff
@@ -639,7 +681,7 @@ class StreamDecoderFilterCallbacks : public virtual StreamFilterCallbacks {
  * - onDestroy is used to cleanup all pending filter resources like pending http requests and
  * timers.
  */
-class StreamFilterBase {
+class StreamFilterBase : public StreamFilterNameHepler {
```

Comment:
The inheritance here is complicated enough as it is. Are all these changes required? Couldn't we just expose name as a function on StreamFilterBase, why does it need to be a base class?

Reply:
It's ok. ​My intention is to have a single helper to implement all name related methods.